### DISCUSSION TITLE
How to scan received items in the app?

### DISCUSSION THREAD

**Discussion by gabrielWilcox** [inventree/InvenTree#8446] [Q&A]
Hi everyone!
I'm trying to use InvenTree in our company to manage our inventory but I can't seem to figure out how to scan received items in the InvenTree app. I tried scanning the InvenTree-generated barcode of the supplier-part but I always get the message "No match for supplier barcode". Am I scanning the wrong barcode? If so, which barcode do I need to scan?

**Comment by SchrodingersGat** [comment]
@gabrielWilcox debugging barcode scanning has proved to be *quite difficult* :) 

So, you can have the server log any scanned barcode and retain scan history in the database. This new feature will be available in 0.17.0 (so if you're currently on the "stable" version, this won't be yet available).

> Am I scanning the wrong barcode? If so, which barcode do I need to scan?

I'm not sure how we can determine that without you providing any information on what barcode you are *actually* scanning, and how it is linked into the system. It sounds like  you are trying to receive items against a purchase order?

**Reply by gabrielWilcox** [reply]
@SchrodingersGat  Thanks for the reply!
Yes I'm trying to receive items against a purchase order with the barcode-function but no matter which barcode I try to scan I always get the error "Not match for supplier barcode". I tried looking it up but couldn't find any info on this. Can you tell me how I can receive items against a purchase order?

**Reply by SchrodingersGat** [reply]
You can only receive items whose barcode contains sufficient information to identify the item within the database. Some suppliers (such as mouser and digikey) are supported so far - but you still need to have the matching supplier part numbers, order codes, etc in the system
